src/update-feeds-loops/update-feeds-loops.ts

```diff
@@ -226,13 +226,11 @@ export const runUpdateFeeds = async (providerName: string, chain: Chain, chainId
 
         // Wait for all the batches to be processed and print stats from this run.
         const processedBatches = await Promise.all([
-          new Promise<Awaited<ReturnType<typeof processBatch>>>((resolve, reject) => {
-            return processFirstBatchPromise.then((result) => {
-              // eslint-disable-next-line promise/always-return
-              if (isError(result)) reject(result);
-              else resolve(result);
-            });
-          }),
+          (async (): Promise<Awaited<ReturnType<typeof processBatch>>> => {
```

Comment:
I made these changes in response to an issue caused by upgrading @total-typescript/ts-reset from version 0.5.1 to 0.6.1, which resulted in TypeScript compilation errors. The update introduced stricter type-checking that required a more explicit approach to handling types and promises.

This immediately invoked function approach also improves code readability.

Reply:
TBH: I'm not a huge fan of IIFE in regards to readability.

Reply:
I see. So, would you prefer to use previous approach or do you have better suggestion on this? I can change this.

Reply:
Nah, I think it's not worth to change this. I think I'd prefer the previous approach (but not sure what was the TS issue there).

Reply:
I mostly wanted express my opinion on that.

Reply:
> but not sure what was the TS issue there

The error was for this part: 

```
error: src/update-feeds-loops/update-feeds-loops.ts:230:45 - error TS2349: This expression is not callable.
  Each member of the union type '{ <TResult1 = { signedApiUrlsFromConfig: string[]; signedApiUrlsFromContract: string[]; beaconIds: 0x${string}[]; successCount: number; errorCount: number; }, TResult2 = never>(onfulfilled?: ((value: { signedApiUrlsFromConfig: string[]; signedApiUrlsFromContract: string[]; beaconIds: 0x${string}[]; successCount:...' has signatures, but none of those signatures are compatible with each other.

230             return processFirstBatchPromise.then((result) => {
                                                ~~~~

src/update-feeds-loops/update-feeds-loops.ts:230:51 - error TS7006: Parameter 'result' implicitly has an 'any' type.

230             return processFirstBatchPromise.then((result) => {
                                                      ~~~~~~
```

Reply:
Mhh, strange. Just from the looks of it I don't know what is wrong. But I think you're solution is good OK, so wouldn't bother investigating further.